ang/civicase/ActivityCard.js

```diff
@@ -1,56 +1,103 @@
+     *
+     * @param {object} activity
+     */
+    $scope.markCompleted = function (activity) {
+      return crmApi([['Activity', 'create', {id: activity.id, status_id: activity.is_completed ? 'Scheduled' : 'Completed'}]])
+        .then(function (data) {
+          if (!data[0].is_error) {
+            activity.is_completed = !activity.is_completed;
+          }
+        });
     };
 
-    $scope.star = function(act) {
-      act.is_star = act.is_star === '1' ? '0' : '1';
+    /**
+     * Star/Favourite an activity
+     *
+     * @param {object} $event
+     * @param {object} activity
+     */
+    $scope.star = function ($event, activity) {
+      $event.stopPropagation();
+      activity.is_star = activity.is_star === '1' ? '0' : '1';
       // Setvalue api avoids messy revisioning issues
-      $scope.refresh([['Activity', 'setvalue', {id: act.id, field: 'is_star', value: act.is_star}]]);
+      $scope.refresh([['Activity', 'setvalue', {id: activity.id, field: 'is_star', value: activity.is_star}]]);
     };
 
-    $scope.deleteActivity = function(activity, dialog) {
+    /**
+     * Delete an activity
+     *
+     * @param {object} activity
+     * @param {jQuery} dialog - the dialog which should be closed once deletion is over
+     */
+    $scope.deleteActivity = function (activity, dialog) {
       CRM.confirm({
-          title: ts('Delete Activity'),
-          message: ts('Permanently delete this %1 activity?', {1: activity.type})
-        })
-        .on('crmConfirm:yes', function() {
+        title: ts('Delete Activity'),
+        message: ts('Permanently delete this %1 activity?', {1: activity.type})
+      })
+        .on('crmConfirm:yes', function () {
           $scope.refresh([['Activity', 'delete', {id: activity.id}]]);
           if (dialog && $(dialog).data('uiDialog')) {
             $(dialog).dialog('close');
           }
         });
     };
 
-    $scope.viewInPopup = function($event, activity) {
+    /**
+     * View an activity details in the popup
+     *
+     * @param {object} $event
+     * @param {object} activity
+     */
+    $scope.viewInPopup = function ($event, activity) {
       if (!$event || !$($event.target).is('a, a *, input, button, button *')) {
         var context = activity.case_id ? 'case' : 'activity';
         var form = CRM.loadForm(CRM.url('civicrm/activity', {action: 'view', id: activity.id, reset: 1, context: context}))
-          .on('crmFormSuccess', function() {
+          .on('crmFormSuccess', function () {
             $scope.refresh();
           })
-          .on('crmLoad', function() {
-            $('a.delete.button').click(function() {
+          .on('crmLoad', function () {
+            $('a.delete.button').click(function () {
               $scope.deleteActivity(activity, form);
+
               return false;
             });
           });
       }
     };
 
-    $scope.moveCopyActivity = function(act, op) {
+    $scope.moveCopyActivity = function (act, op) {
```

Comment:
Can we document this function too ?

Reply:
Same as above, I have not touched it because I have not worked on it. In future when we touch this feature, we should update the doc.